### ISSUE TITLE
Arbritrary machine moves after restarting LitePlacer

### ISSUE THREAD

**Issue by mrandt** [jkuusama/LitePlacer-ver2#21]
After LitePlacer software had crashed and killed via taskmanager, machine moves arbritrarily after restart of LitePlacer. I think TinyG is not correctly reinitialized.
Hardware reset on TinyG and reconnecting from LitePlacer cures the problem.

**Comment by thethereza** [comment]
yeah, i've noticed this and not sure why it does it either. i issue the command pump off vacuum off which does a small movement (to fix some bug that i don't understand according to the source) and i think this somehow makes it move a lot.  needs more investigation to find the root cause.

**Comment by jkuusama** [comment]
> the command pump off vacuum off which does a small movement (to fix some bug that i don't understand according to the source

Discussion here: https://www.synthetos.com/topics/file-not-open-error/

Summary: Due a bug in TinyG, if there is an M command and machine status settings in the queue, the machine settings go through only after a move.

**Comment by mrandt** [comment]
I am aware that we need to make a move to switch vacuum. 
But my scenario is different - steps to replicate:
1. Start LitePlacer (v2)
2. Move machine somewhere in the middle of the table (precaution)
3. Kill LitePlacer via taskmanager - this happened to me when application crashed but that is somewhat less deterministic to reproduce
4. Start LitePlacer again
-> machine moves slowly into a direction and does not stop until TinyG is reset

I think it might be related to initilization when connecting to TinyG, maybe serial buffers are not flushed, commands are still in queue of TinyG, command sequence does not fit TinyG status or the like.

**Comment by thethereza** [comment]
What did he mean by set and reset in this context

The machining state is correctly set when the machine is moving, and also
correctly reset when the movement is finished. The funny thing (aka bug) is
that it is also set when M8 (or any other M command) is executed after the
movement stops. It is only reset when another movement is started and
finished.

On Tuesday, July 14, 2015, Juha Kuusama notifications@github.com wrote:

> > the command pump off vacuum off which does a small movement (to fix some
> > bug that i don't understand according to the source
> 
> Discussion here: https://www.synthetos.com/topics/file-not-open-error/
> 
> Summary: Due a bug in TinyG, if there is an M command and machine status
> settings in the queue, the machine settings go through only after a move.
> 
> > thethereza <notifications@github.com
> > <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> kirjoitti
> > 14.7.2015 kello 21.25:
> > 
> > yeah, i've noticed this and not sure why it does it either. i issue the
> > command pump off vacuum off which does a small movement (to fix some bug
> > that i don't understand according to the source) and i think this somehow
> > makes it move a lot. needs more investigation to find the root cause.
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jkuusama/LitePlacer-ver2/issues/21#issuecomment-121495412
> .

## 

-Reza
